src/components/dialog-parameters/load-flow-parameter.js

```diff
@@ -0,0 +1,459 @@
+            getLoadFlowProvider(studyUuid)
+                .then((provider) => {
+                    // if provider is not defined or not among allowed values, it's set to default value
+                    if (!(provider in LF_PROVIDER_VALUES)) {
+                        setLoadFlowProviderToDefault();
+                    } else {
+                        setLfProvider(provider);
+                    }
+                })
+                .catch((errorMessage) =>
+                    displayErrorMessageWithSnackbar({
+                        errorMessage: errorMessage,
+                        enqueueSnackbar: enqueueSnackbar,
+                        headerMessage: {
+                            headerMessageId: 'getLoadFlowProviderError',
+                            intlRef: intlRef,
+                        },
+                    })
+                );
+        }
+    }, [studyUuid, enqueueSnackbar, intlRef, setLoadFlowProviderToDefault]);
+
+    function MakeAdvancedParameterButton(showOpenIcon, label, callback) {
+        return (
+            <>
+                <Grid item xs={12} className={classes.advancedParameterButton}>
+                    <Button
+                        variant="outlined"
+                        startIcon={<SettingsIcon />}
+                        endIcon={
+                            showOpenIcon ? (
+                                <CheckIcon style={{ color: 'green' }} />
+                            ) : undefined
+                        }
+                        onClick={callback}
+                    >
+                        <FormattedMessage id={label} />
+                    </Button>
+                </Grid>
+            </>
+        );
+    }
+
+    function makeComponentsFor(defParams, params, setter) {
+        return Object.keys(defParams).map((key) => (
+            <Grid container spacing={1} paddingTop={1} key={key}>
+                {makeComponentFor(defParams[key], key, params, setter)}
+                <LineSeparator />
+            </Grid>
+        ));
+    }
+
+    function makeComponentFor(defParam, key, lfParams, setter) {
+        if (defParam.type === TYPES.bool) {
+            return MakeSwitch(lfParams[key], defParam.description, (ev) =>
+                setter({ ...lfParams, [key]: ev.target.checked })
+            );
+        } else if (defParam.type === TYPES.enum) {
+            return MakeDropDown(
+                lfParams[key],
+                defParam.description,
+                defParam.values,
+                (ev) => setter({ ...lfParams, [key]: ev.target.value })
+            );
+        } else if (defParam.type === TYPES.countries) {
+            return MakeCountrySelector(
+                lfParams[key],
+                defParam.description,
+                (newValues) => {
+                    setter({ ...lfParams, [key]: [...newValues] });
+                }
+            );
+        }
+    }
+
+    function MakeCountrySelector(value, label, callback) {
```

Comment:
const CountrySelector = ( {value, label, callback}) =>

Reply:
it's corrected